core/property/registry.go

```diff
@@ -2,31 +2,89 @@ package property
 
 import (
 	"fmt"
-	"sync"
-
-	"github.com/TIBCOSoftware/flogo-lib/core/data"
+	"github.com/TIBCOSoftware/flogo-lib/config"
 	"github.com/TIBCOSoftware/flogo-lib/logger"
+	"os"
+	"reflect"
+	"regexp"
+	"strings"
+	"sync"
 )
 
 var (
-	props = make(map[string]string)
+	props = make(map[string]interface{})
 	mut   = sync.RWMutex{}
+	regex = regexp.MustCompile(config.GetPropertyDelimiterFormat())
+	// Default Resolver
+	resolver Resolver = &DefaultResolver{}
 )
 
+// Resolve value sourced from Enviornment variable or any other configuration management services
+type Resolver interface {
+	//Resolve value for given name
+	Resolve(name string) interface{}
```

Comment:
I think we can use the ok bool pattern, instead of another function in the interface for CanResolve.
Something like: 
Resolve(name string) (interface{}, bool)

And we will discuss today about the type interface{}